### ISSUE TITLE
Release the ReportViewer source code under a permissive license

### ISSUE THREAD

**Issue by HeinziAT** [dotnet/aspnetcore#51187]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

It has been stated multiple times (e.g. [here](https://github.com/dotnet/aspnetcore/issues/39912)) in this repository and at other places that the ReportViewer (RDLC) rendering engine for .NET Framework will not be ported to .NET (Core), that Microsoft has no plans to develop the RDLC engine further and that you want people to switch to a could-based solution (Power BI) instead.

That's fine. However, there's a lot of people outside of Microsoft who *do* want to port and/or extend the RDLC engine. I'm sure you are aware of the (non-Microsoft) github repos and nuget packages offering a .NET (Core) compatible version of the RDLC engine.

The only thing that's preventing us (and probably a lot of other companies) from using them is that these packages are in a legal grey zone: They have been created by decompiling and modifying closed-source Microsoft software. Depending on your jurisdiction, redistributing them might not be legal.

### Describe the solution you'd like

Thus, my feature request would be to release the ReportViewer sources with a permissive license, or, if that is not possible, at least officially state that it's OK to use those "unofficial" .NET Core ports. 

### Additional context

(Yes, I'm aware that ReportViewer is maintained by the SQL Server team, but the copyright is held by the company Microsoft, not by a specific team inside Microsoft.)

**Comment by gfoidl** [comment]
@HeinziAT thanks for your interest, but https://github.com/dotnet/aspnetcore/issues/39912#issuecomment-1027013004 pp. still applies.

> the copyright is held by the company Microsoft, not by a specific team inside Microsoft

Yep, but unfortunately this repo here is the wrong place for this as it's for ASP.NET Core.
(Maybe it can be moved to e.g. Winforms repo (and I don't have the rights to do so))

**Comment by HeinziAT** [comment]
I get that, and I thought a lot about which repo *is* the correct one: There is no SSRS repo, github.com/microsoft/dotnet is no longer maintained, and Winforms is also not right, since I don't particularly care about the UI (WinForms, WebForms) components of ReportViewer. The rendering engine that transforms an RDLC file + a list of data objects into a PDF is the interesting part.

The thing is: The unavailability of an RDLC renderer is a .NET Framework -> .NET migration blocker, independent of the UI technology used. In the end, I settled for the aspnetcore repo since this is where I found the most past activity regarding this issue. Apparently, it's more of a ASP.NET -> ASP.NET Core migration blocker than a .NET FW WinForms -> .NET WinForms blocker. :-)

I also considered dotnet/runtime (responsible for the BCL), but my suggestion is not to add the RDLC rendering engine to the BCL (it's clear that this won't happen). It's about open-sourcing a Microsoft library *which currently doesn't have a repo*, but which would (indirectly) benefit ASP.NET Core (as well as WinForms, WPF and UI-less services).

So, yes, if you have a more suitable repo, I'll be more than happy to migrate the issue.
